src/arena.rs

```diff
@@ -63,7 +64,7 @@ impl<T> ArenaAllocator<T> {
     fn cleanup<B, A>(&mut self, owner: &mut A, device: &B::Device)
     where
         B: Backend,
-        T: Block<B>,
+        T: Block<Memory = B::Memory>,
         A: MemoryAllocator<B, Block = T>,
```

Comment:
could do this instead of having `T`:
```rust
where
    A::Block: Block<Memory = B::Memory>
```

Reply:
`A::Block: Block<Memory = B::Memory>` is always held for `A: MemoryAllocator<B>`.
While `T` is a type parameter of `ArenaAllocator` itself. And it must be the type of superblock (`A::Block`)